src/compose/ComposeBox.js

```diff
@@ -217,6 +219,36 @@ class ComposeBox extends PureComponent<Props, State> {
     }
   };
 
+  replaceMessageTextAtRange = (text: string, beg: number, end: number) => {
+    const { message } = this.state;
+    this.setMessageInputValue(message.slice(0, beg) + text + message.slice(end, message.length));
+    return [beg, beg + text.length];
+  };
+
+  insertAttachment = async (attachments: [DocumentPickerResponse]) => {
+    this.setState(({ isUploading }) => ({
+      isUploading: !isUploading,
+    }));
+
+    const { _, auth } = this.props;
+    let [beg, end] = [undefined, this.state.selection.start];
+    for (let i = 0; i < attachments.length; i++) {
+      const fileName = attachments[i].name ?? `attachment-${i + 1}`;
+      [beg, end] = this.replaceMessageTextAtRange(`[Uploading ${fileName}...]()\n`, end, end);
```

Comment:
We'll also want this to be translated. See previous comment for link to our docs on how.

Reply:
same concern as [above](https://github.com/zulip/zulip-mobile/pull/4590#discussion_r622597382)